### PR TITLE
Add Kubernetes support

### PR DESCRIPTION
**Pull request by keur**

### CODE REVIEW

File: kubernetes/ocfweb-web.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    matchLabels:
+      app: ocfweb-web
+  template:
+    metadata:
+      labels:
+        app: ocfweb-web
+    spec:
+      containers:
+        - name: ocfweb-web-deploy
+          image: "docker.ocf.berkeley.edu/ocfweb-web:keur-1"
```
Comment (kpengboy) [inline kubernetes/ocfweb-web.yaml]: This should be changed to the real value before merging, no? (Same for the other files)

Comment (keur) [reply kubernetes/ocfweb-web.yaml]: Yeah thanks

File: kubernetes/ocfweb-worker.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: ocfweb-worker-service
+spec:
+  selector:
+    app: ocfweb-worker
+  ports:
+    - port: 80
+      targetPort: 8000
```
Comment (dkess) [inline kubernetes/ocfweb-worker.yaml]: I don't think this needs a `Service`.

Comment (keur) [reply kubernetes/ocfweb-worker.yaml]: TrUwU

**Comment by keur** [comment]
@jvperrin @chriskuehl if there is a better way to make `pre-commit` ignore multiple yaml streams in the same file, while also running everything else in the `check-yaml` hook, that would be ideal.

File: kubernetes/ocfweb-web.yaml.erb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            path: /opt/share/kubernetes/secrets/ocfweb
+            type: Directory
+        - name: ocfweb-ocf
+          hostPath:
+            path: /etc/ocf
+            type: Directory
+      dnsPolicy: ClusterFirst
+      dnsConfig:
+        searches:
+          - "ocf.berkeley.edu"
```
Comment (cg505) [inline kubernetes/ocfweb-web.yaml.erb]: @keur why is this necessary?

**Comment by cg505** [comment]
also why are the tests failing?

File: kubernetes/ocfweb-web.yaml.erb [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      targetPort: 8000
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ocfweb-web-deployment
+  labels:
+    app: ocfweb-web
+spec:
+  replicas: 1
```
[the remark this answers is not in the document]
Comment (keur) [reply kubernetes/ocfweb-web.yaml.erb]: good catch. this was a testing setting back when the cluster was on lightly provisioned VMs

File: kubernetes/ocfweb-web.yaml.erb (hunk 3 above)
Comment (keur) [reply kubernetes/ocfweb-web.yaml.erb]: It's when we use OCF hostnames (not fqdn). But it doesn't appear to be necessary. After glancing over the code I can't seem to find an instance where we use one.

**Comment by keur** [comment]
@cg505 just needed a rebase

File: kubernetes/ocfweb-web.yaml.erb (hunk 3 above)
Comment (dkess) [reply kubernetes/ocfweb-web.yaml.erb]: I'd keep it just in case, it's probably used in various places in ocflib.

Comment (cg505) [reply kubernetes/ocfweb-web.yaml.erb]: Do we need `dnsPolicy: ClusterFirst` though?

File: kubernetes/ocfweb-worker.yaml.erb [hunk 5]
```diff
@@ -0,0 +1,48 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ocfweb-worker-deployment
+  labels:
+    app: ocfweb-worker
+spec:
+  replicas: 1
```
Comment (cg505) [inline kubernetes/ocfweb-worker.yaml.erb]: :thinking:

Comment (keur) [reply kubernetes/ocfweb-worker.yaml.erb]: (🤔)[https://github.com/ocf/services/blob/090b1ef6c09d82b0f1713d47ffe7692a2fb50356/apps/ocfweb/worker/app.yaml#L8]

Comment (keur) [reply kubernetes/ocfweb-worker.yaml.erb]: also, i think the reasoning is since it just writes stuff to redis, even if the worker goes down the site is still live. also having multiple workers doing the same computations and persisting them to redis (one potentially overwriting the other) is not particularly helpful.

File: kubernetes/ocfweb-web.yaml.erb (hunk 3 above)
Comment (keur) [reply kubernetes/ocfweb-web.yaml.erb]: `ClusterFirst` is the default `dnsPolicy` so no we don't need it because it doesn't change anything. For clarity, when using `dnsConfig`, I always include it. I'm not particularly attached if you want to remove it.

File: kubernetes/ocfweb-static.yaml.erb [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+          ports:
+            - containerPort: 8000
+---
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: vhost-static-ingress
+spec:
+  rules:
+    - host: static.www.ocf.berkeley.edu
```
Comment (abizer) [inline kubernetes/ocfweb-static.yaml.erb]: static.www.ocf?

Comment (cg505) [reply kubernetes/ocfweb-static.yaml.erb]: wait...

**Comment by dkess** [comment]
Is this ready for another review?

**Comment by fydai** [comment]
Yes

File: kubernetes/ocfweb-web.yaml.erb (hunk 3 above)
Comment (fydai) [reply kubernetes/ocfweb-web.yaml.erb]: in particular, logs are sent to "syslog"
